### ISSUE TITLE
Use return value instead of exception?

### ISSUE THREAD

**Issue by pstutz** [lmdbjava/lmdbjava#39]
Based on the special case handling of the `MDB_NOOVERWRITE` put flag I assume that this is a supported way of retrieving an existing value when putting.  The ability to put and know the entry already existed is really useful for my use case, where I build an index bottom-up, cascading until an entry already exists (MDB_DUPSORT on the Dbi, MDB_NODUPDATA on the put):
```scala
    try {
      db.put(txn, key, value, MDB_NODUPDATA)
      true
    } catch {
      case e: KeyExistsException =>
        false
      case NonFatal(e) => ...
    }
```
The issue with this solution is that my index loading time is dominated by string formatting due to exception initialisations.

I was wondering if it might be possible to instead use a return value to communicate if a key/value already existed. I [tested it](https://github.com/pstutz/lmdbjava/commit/b845dd4c0466ace6871f6fce600a9f8be45dd297) and the performance was much improved.

Is there a better approach to achieving my goal? Would you be interested in a pull request for an approach where the return value is used to communicate extra information instead of an exception?

**Comment by benalexau** [comment]
Performance is one of LmdbJava's [priorities](https://github.com/lmdbjava/lmdbjava/blob/master/src/main/java/org/lmdbjava/package-info.java), so +1 from me. I'd just ask you to also apply the same logic to `Cursor` for the sake of consistency and submit a PR.

More generally I wonder if there are other return type `void` situations where a return value would be useful?

**Comment by pstutz** [comment]
For `put` addressed by #40. Thanks a lot!
